dhis-2/dhis-api/src/main/java/org/hisp/dhis/icon/CustomIconStore.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2004-2023, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.icon;
+
+import java.util.List;
+
+import org.hisp.dhis.fileresource.FileResource;
+import org.hisp.dhis.user.User;
+
+public interface CustomIconStore
+{
+    /**
+     * Returns a custom icon that contains a given key
+     *
+     * @param key of the icon
+     * @return the custom icon matching the key, or null instead
+     */
+    CustomIcon getIconByKey( String key );
+
+    /**
+     * Returns a list of custom icons that contain all the specified keywords
+     *
+     * @param keywords the icon needs to contain
+     * @return the list of custom icons that contain all the keywords
+     */
+    List<CustomIcon> getIconsByKeywords( String[] keywords );
+
+    /**
+     * Gets all custom icons present in the database
+     *
+     * @return a list containing all icons
+     */
+    List<CustomIcon> getAllIcons();
```

Comment:
I would be consistent in the signatures across service and store

`List<CustomIcon> getAllIcons();` (Store), `Collection<Icon> getIcons();` (Service)